AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthenticationProviderAdapter+Password.swift

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSMobileClient
+
+extension AuthenticationProviderAdapter {
+
+    func resetPassword(request: AuthResetPasswordRequest,
+                       completionHandler: @escaping (Result<AuthResetPasswordResult, AmplifyAuthError>) -> Void ) {
+
+        let clientMetaData = (request.options.pluginOptions as? AWSAuthResetPasswordOptions)?.metadata ?? [:]
+        awsMobileClient.forgotPassword(
+            username: request.username,
+            clientMetaData: clientMetaData) { result, error in
+                if let error = error {
+                    let authError = AuthErrorHelper.toAmplifyAuthError(error)
+                    completionHandler(.failure(authError))
+                    return
+                }
+                guard let result = result else {
+                    // This should not happen, return an unknown error.
+                    let error = AmplifyAuthError.unknown("Could not read result from resetPassword operation")
+                    completionHandler(.failure(error))
+                    return
+                }
+                guard let codeDeliveryDetails = result.codeDeliveryDetails?.toAuthCodeDeliveryDetails() else {
+                    let authResetPasswordResult = AuthResetPasswordResult(isPasswordReset: true, nextStep: .done)
+                    completionHandler(.success(authResetPasswordResult))
+                    return
```

Comment:
Reading this code makes me believe that we can call `resetPassword()`, and then not have to call `confirmResetPassword` (with a new password and confirmationCode).  Am i reading this right?  Could you help me better understand what use case this code is for?

Reply:
Yes, you are right. There is confirmResetPassword only if you have codeDeliveryDetails otherwise resetPassword is complete.